### PR TITLE
Tweening engine

### PR DESCRIPTION
**Pull request by luismoyano**
Tweening engine working to do tweening animations.

It supports:
- Tween transform 2d position, rotation and scale
- Tween Image Color
-  Supports multiple easings
- Supports loops (Restart and Yoyo)

### CODE REVIEW

File: Engine/Animation/Tween/Tween.cpp [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+float Tween::NormalizedElapsedTime()
+{
+	//A = start_time / 0
+	//B = start_time + duration / 1
+
+	//C = current_time / y
+
+	float normalized_time =  (current_time - start_time) * (1.0f / duration);
+
+	return normalized_time;
```
Comment (Dartemiss) [inline Engine/Animation/Tween/Tween.cpp]: This method should be const and you can return directly the line 153

File: Engine/Animation/Tween/Tween.cpp [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+		case Tween::COLOR:
+			image->SetColor(initial_color);
+			break;
+		}
+	}
+}
+
+float Tween::Linear(float t) {
+	return t;
+}
```
Comment (Dartemiss) [inline Engine/Animation/Tween/Tween.cpp]: Can be const

File: Engine/Animation/Tween/Tween.cpp [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+
+float Tween::EaseInOutBack(float t) {
+	float s = 1.70158f * 1.525f;
+	//float s2 = s * 1.525f;
+	t *= 2;
+	if (t < 1) return 1.0 / 2 * (t*t*((s + 1)*t - s));
+	float postFix = t -= 2;
+	return 1.0 / 2 * ((postFix)*t*((s + 1)*t + s) + 2);
+}
```
Comment (Dartemiss) [inline Engine/Animation/Tween/Tween.cpp]: All of them can be const

File: Engine/Animation/Tween/TweenSequence.cpp [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+	return this;
+}
+
+TweenSequence * TweenSequence::Insert(float insert_time, Tween* new_tween)
+{
+	new_tween->start_time = insert_time;
+	tweens.push_back(new_tween);
+
+	return nullptr;
```
Comment (Dartemiss) [inline Engine/Animation/Tween/TweenSequence.cpp]: Why are u returning a nullptr?

File: Engine/Animation/Tween/TweenSequence.cpp [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+	state = TweenSequenceState::PAUSED;
+
+	return this;
+}
+
+void TweenSequence::Update(float dt)
+{
+	if (state != TweenSequenceState::PLAYING) return;
+
+	current_time += dt / 1000.0f;
```
Comment (Dartemiss) [inline Engine/Animation/Tween/TweenSequence.cpp]: It is better to multiply by 0.001

**Review by Dartemiss — APPROVED** [review]
Good Job! Very clean code :D Just a few comments. Cool feature

File: Engine/Animation/Tween/Tween.cpp (hunk 1 above)
Comment (luismoyano) [reply Engine/Animation/Tween/Tween.cpp]: Done!

File: Engine/Animation/Tween/Tween.cpp (hunk 2 above)
Comment (luismoyano) [reply Engine/Animation/Tween/Tween.cpp]: Done!

File: Engine/Animation/Tween/TweenSequence.cpp (hunk 4 above)
Comment (luismoyano) [reply Engine/Animation/Tween/TweenSequence.cpp]: My mistake, fixed!

File: Engine/Animation/Tween/TweenSequence.cpp (hunk 5 above)
Comment (luismoyano) [reply Engine/Animation/Tween/TweenSequence.cpp]: Done!

File: Engine/Module/ModuleUI.cpp [hunk 6]
```diff
@@ -8,8 +9,10 @@
 #include "Component/ComponentText.h"
 #include "Component/ComponentButton.h"
 
+#include "Main/Application.h"
 #include "Main/Globals.h"
 #include "Main/GameObject.h"
+#include "Animation/Tween/LOTween.h"
```
Comment (yalania) [inline Engine/Module/ModuleUI.cpp]: This includes are needed?

**Review by yalania — APPROVED** [review]
Looks awesome, it would be nice to be able to do timelines from the editor and add them to the state machine later. Super nice job!

File: Engine/Module/ModuleUI.cpp (hunk 6 above)
Comment (luismoyano) [reply Engine/Module/ModuleUI.cpp]: My mistake, removed already

**Pull request merged**